src/decoder/mod.rs

```diff
@@ -542,11 +550,25 @@ pub enum DecoderError {
 
 impl fmt::Display for DecoderError {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "{}", self.to_string())
+        let text = match self {
+            DecoderError::UnrecognizedFormat => "Unrecognized format",
+            #[cfg(feature = "symphonia")]
+            DecoderError::IoError(msg) => &msg[..],
+            #[cfg(feature = "symphonia")]
+            DecoderError::DecodeError(msg) => msg,
+            #[cfg(feature = "symphonia")]
+            DecoderError::LimitError(msg) => msg,
+            #[cfg(feature = "symphonia")]
+            DecoderError::ResetRequired => "Reset required",
+            #[cfg(feature = "symphonia")]
+            DecoderError::NoStreams => "No streams",
+        };
+        write!(f, "{}", text)
     }
 }
 
 impl Error for DecoderError {
+    /*
```

Comment:
Can you instead completely remove this?

Reply:
Yes.  That's my mistake - I forgot to review my changes after I tested them.